src/OpenLoco/src/Paint/Paint.h

```diff
@@ -51,6 +51,57 @@ namespace OpenLoco::Paint
     // Handy array for converting a bit index to the flag
     constexpr std::array<SegmentFlags, 9> kSegmentOffsets = { SegmentFlags::x0y0, SegmentFlags::x2y0, SegmentFlags::x0y2, SegmentFlags::x2y2, SegmentFlags::x1y1, SegmentFlags::x1y0, SegmentFlags::x0y1, SegmentFlags::x2y1, SegmentFlags::x1y2 };
 
+    constexpr uint8_t rotl4bit(uint8_t val, uint8_t rotation)
+    {
+        return ((val << rotation) | (val >> (4 - rotation))) & 0xF;
+    }
+
+    constexpr uint8_t rotr4bit(uint8_t val, uint8_t rotation)
+    {
+        return ((val >> rotation) | (val << (4 - rotation))) & 0xF;
+    }
```

Comment:
I think its time to move this into Numerics.hpp, I believe I've seen this before.

Reply:
You saw it before because i moved it in this pr. Should have put it in numerics doh.

Reply:
Right, its at the very bottom so I didn't quite realize it.